File: shruti170901_Leetcode/Split a String in Balanced Strings.cpp
#include <bits/stdc++.h>
#include <vector>
#include <iostream>
#include <string>
#include <map>
#include <set>
#include <unordered_map>
#include <algorithm>
using namespace std;

// https://leetcode.com/problems/split-a-string-in-balanced-strings/

class Solution {
public:
    int balancedStringSplit(string s) {
        int cnt=0, temp=0;
        for(int i=0;i<s.size();i++){
            if(s[i]=='L') temp++;
            else temp--;
            if(temp==0) cnt++;
        }
        return cnt;
    }
};

File: Architgupta001_LeetCode-Solutions/PYTHON/Split a String in Balanced Strings.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        c=cl=cr=0
        for i in s:
            if i == "R":
                cr +=1
            elif i == "L":
                cl+=1
            if cr==cl:
                c+=1
        return c